### PR TITLE
Small performance improvements

### PR DESCRIPTION
**Pull request by Iliano101**
Hey there,
I made a few changes that improve the performance, mainly in the HomeScreen. 
Here is the list :
- Added a ProGuard rule to remove all logs from the release build. The Log.w and Log.e are not removed by default by the compiler.
- Added @Immutable to some classes. This has the effect of telling Android Compose that these classes don't need to be recomposed as often, improving performance.
- Added a Key to all items enumeration in the HomeScreen. This also has the effect of telling Android Compose that the items only need to be recomposed if they are different.

In total, this improves the performance in the HomeScreen and in the app as a whole. There is still some lag, but it is smaller.

### CODE REVIEW

**Comment by maxrave-dev** [comment]
Can you fix conflict with my branch

**Comment by Iliano101** [comment]
Alright, it won't be long

**Comment by Iliano101** [comment]
@maxrave-dev 
Done

File: .idea/runConfigurations.xml
Comment (maxrave-dev) [inline .idea/runConfigurations.xml:1]: Please remove this file

**Comment by Iliano101** [comment]
@maxrave-dev 
Done

**Review by maxrave-dev — APPROVED** [review]

**Comment by maxrave-dev** [comment]
@CodiumAI-Agent /improve

**Comment by maxrave-dev** [comment]
@Iliano101 Thanks for contribute <3

**Pull request merged**